src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +46,69 @@
      * Андрей - 600
      * София - 900</p>
      */
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+    private static DateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy");
+
+    public static String getSalaryInfo(String[] names, String[] data,
+                                       String dateFrom, String dateTo)
+            throws IllegalDateParametersException {
+        if (!checkDate(dateFrom, dateTo)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+        StringBuilder report = new StringBuilder("Отчёт за период " + dateFrom + " - " + dateTo);
+        for (String name : names) {
+            int salary = 0;
+            for (int i = 0; i < data.length; i++) {
+                if (data[i].contains(name)) {
+                    String[] dataRecord = data[i].split(" ");
+                    if (checkDate(dateFrom, dataRecord[0]) && checkDate(dataRecord[0], dateTo)) {
+                        salary += Integer.parseInt(dataRecord[2]) * Integer.parseInt(dataRecord[3]);
+                    }
+                }
+            }
+            report.append("\n").append(name).append(" - ").append(salary);
+        }
+
+        return report.toString() + "\n";
     }
+
+    public static boolean checkDate(String dateFrom, String dateTo) {
+        try {
+            Date dateToF = dateFormat.parse(dateTo);
```

Comment:
try to use LocalDate. By the way, it has own methods to check dates, so you don't need to create this one :)